src/structures/Message.js

```diff
@@ -571,8 +571,9 @@ class Message extends Base {
    */
   get pinnable() {
     return (
-      this.type === 'DEFAULT' &&
-      (!this.guild || this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false))
+      ['DEFAULT', 'REPLY', 'APPLICATION_COMMAND', 'THREAD_STARTER_MESSAGE'].includes(this.type) &&
+      (!this.guild || this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false)) &&
+      !this.system
```

Comment:
I'm confused as of why this is needed, when all it does is check whether or not `this.type` is one of a few:

https://github.com/discordjs/discord.js/blob/f57791c5feff1f9808369f8a7f95dab0900c6f0c/src/structures/Message.js#L69-L73

And we're already checking for that type in line 574 from this PR.

Reply:
Sorry I got confused by @SpaceEEC’s comment then. Space can you tell me how you got the error you mentioned above e.g. what message type it was?

Reply:
This is the code in question:
`message.channel.messages.fetch().then(m => m.find(m => m.type == 'THREAD_STARTER_MESSAGE')).then(m => m.pin())`

Reply:
Ooohh okay my bad
Can you show me what a THREAD_STARTER_MESSAGE is or what it looks like btw?

Reply:
It's just an empty message with a reference to the message the thread was created on:
<details>
<summary>THREAD_STARTER_MESSAGE  message</summary>

```js
{
  id: '884893836730572802',
  type: 21,
  content: '',
  channel_id: '884893816438542396',
  author: [Object],
  attachments: [],
  embeds: [],
  mentions: [],
  mention_roles: [],
  pinned: false,
  mention_everyone: false,
  tts: false,
  timestamp: '2021-09-07T20:12:11.209000+00:00',
  edited_timestamp: null,
  flags: 0,
  components: [],
  message_reference: {
    channel_id: 'id',
    guild_id: 'id',
    message_id: '884893816438542396'
  },
  referenced_message: {
    id: '884893816438542396',
    type: 0,
    content: 'test',
    channel_id: 'id',
    author: [Object],
    attachments: [],
    embeds: [],
    mentions: [],
    mention_roles: [],
    pinned: false,
    mention_everyone: false,
    tts: false,
    timestamp: '2021-09-07T20:12:06.371000+00:00',
    edited_timestamp: null,
    flags: 32,
    components: [],
    thread: [Object]
  }
}
```
</details>